### PR TITLE
feat(app,shared-data): get addressable area for legacy fixed trash

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview

provides a special case for the addressable area of the legacy Flex fixed trash found in older protocols. parses the labware key of a protocol analysis and inserts addressable area `moveableTrashA3` if a legacy fixed trash labware is found. moves the `getSimplestDeckConfigForProtocol` function to shared-data and updates references in app and components.

closes RAUT-888

# Test Plan

 - manually verified rendering and functionality
 - updated unit tests

# Changelog

 - Adds a special case for getting the addressable area of the legacy Flex fixed trash for pre-deck config protocols

# Review requests

 - confirm functionality, approach. the idea is to capture all instances of Flex legacy fixed trash and end up with the correct set of addressable areas for a protocol.

# Risk assessment

low

### CODE REVIEW

File: app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/index.tsx [hunk 1]
```diff
@@ -27,22 +27,26 @@ import { SetupModulesMap } from './SetupModulesMap'
...
+  ProtocolAnalysisOutput,
+} from '@opentrons/shared-data'
 
 interface SetupModuleAndDeckProps {
   expandLabwarePositionCheckStep: () => void
   robotName: string
   runId: string
   hasModules: boolean
-  commands: RunTimeCommand[]
+  // TODO: what  type is this
```
Comment (jerader) [inline app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/index.tsx]: is this TODO needed?

File: components/src/hardware-sim/BaseDeck/BaseDeck.tsx [hunk 2]
```diff
@@ -253,7 +253,10 @@ export function BaseDeck(props: BaseDeckProps): JSX.Element {
           }) => {
             if (
               labwareLocation === 'offDeck' ||
-              !('slotName' in labwareLocation)
+              !('slotName' in labwareLocation) ||
+              // do not render flex trash as a labware
+              definition.parameters.loadName ===
+                'opentrons_1_trash_3200ml_fixed'
```
Comment (jerader) [inline components/src/hardware-sim/BaseDeck/BaseDeck.tsx:259]: maybe the comment above can specify that this is for legacy protocols only with a loadLabware trash command.

**Review by b-cooper — APPROVED** [review]
👴 🗑️

**Review by jerader — APPROVED** [review]

File: app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/index.tsx (hunk 1 above)
Comment (brenthagen) [reply app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/index.tsx]: oops, thanks

**Pull request merged**